hazelcast/src/main/java/com/hazelcast/map/impl/MapEntries.java

```diff
@@ -35,31 +35,139 @@
  */
 public final class MapEntries implements IdentifiedDataSerializable {
 
+    private static interface ValueWrapper {
```

Comment:
@LoneRifle, the changes here are required to avoid creating defensive copies as a result of a getAll method call. The current implementation of MapEntries requires the value to be represented as Data, which forces a defensive copy to populate MapEntries. The proposed changes defer the serialization to Data until necessary and leaves as Object as long as possible.

Do you have an alternative suggestion on how to avoid defensive copies on getAll operations?

Do you have benchmarks to measure the impacts you are concerned about?